internal/pubsub/drivertest/drivertest.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package drivertest provides a conformance test for implementations of
+// driver.
+package drivertest
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/google/go-cloud/internal/pubsub"
+)
+
+// Harness descibes the functionality test harnesses must provide to run
+// conformance tests.
+type Harness interface {
+	// MakeTopicDriver creates a Topic and associated Subscription to test.
```

Comment:
Pass a `context.Context` to `Make` functions, don't save the `ctx` from `HarnessMaker`.

Reply:
Done.